mapbox_vector_tile/encoder.py

```diff
@@ -56,13 +56,28 @@ def transform(shape, func):
     raise ValueError('Unknown geometry type, "%s"' % shape.type)
 
 
+def on_invalid_geometry_raise(shape):
+    raise Exception('Invalid geometry: %s' % shape.wkt)
+
+
+def on_invalid_geometry_ignore(shape):
+    return None
+
+
+def on_invalid_geometry_make_valid(shape):
+    if shape.type in ('Polygon', 'MultiPolygon'):
+        shape = shape.buffer(0)
```

Comment:
You're right, it does seem to preserve the winding order. I added a test for that behavior in 4e356b6.

It's also true that the values became floating point after the `buffer`. We do `round` them again, but like you hinted at, I suppose it is possible for this to alter the winding order again.

Would we need to re-quantize though? I would assume that we would just need to do the rest of the chain, namely round, ensure the correct winding order, and then handle invalid geometries again. And maybe we try this 5 times or so before dropping the feature?